packages/stark-core/src/session/services/session.service.ts

```diff
@@ -0,0 +1,411 @@
+		this.idle.setInterrupts(DEFAULT_INTERRUPTSOURCES);
+		this.startIdleService();
+		this.startKeepaliveService();
+	}
+
+	protected configureIdleService(): void {
+		// ensuring that the session timeout values are valid to prevent errors while setting the idle value
+		if (this.appConfig.sessionTimeoutWarningPeriod >= this.appConfig.sessionTimeout) {
+			const errorMsg: string = starkSessionServiceName + ": " + STARK_APP_CONFIG + " constant is not valid.\n\n" +
+				"- sessionTimeoutWarningPeriod cannot be equal or higher than sessionTimeout\n";
+
+			throw new Error(errorMsg);
+		}
+
+		// seconds before the user is considered to be idle (should be calculated subtracting the timeout warning period)
+		this.idle.setIdle(this.appConfig.sessionTimeout - this.appConfig.sessionTimeoutWarningPeriod);
+		// seconds before the session times out and the timeout warning event should be emitted
+		this.idle.setTimeout(this.appConfig.sessionTimeoutWarningPeriod);
+		// sets the default interrupts (clicks, scrolls, touches to the document)
+		this.idle.setInterrupts(DEFAULT_INTERRUPTSOURCES);
+
+		this.idle.onIdleStart.subscribe(
+			() => this.logger.warn(starkSessionServiceName + ": the user has gone idle")
+		);
+		this.idle.onIdleEnd.subscribe(
+			() => {
+				this.logger.info(starkSessionServiceName + ": the user is no longer idle");
+				if (this.countdownStarted) {
+					this.countdownStarted = false;
+					// dispatch action so an effect can run any logic if needed
+					this.store.dispatch(new SessionTimeoutCountdownStop());
+				}
+			}
+		);
+
+		this.idle.onTimeout.subscribe(
+			() => {
+				this.logger.warn(starkSessionServiceName + ": the user session has timed out!");
+				// dispatch action so an effect can run any logic if needed
+				this.store.dispatch(new SessionTimeoutCountdownFinish());
+				this.logout();
+				this.routingService.navigateTo(starkSessionExpiredStateName);
+			}
+		);
+		this.idle.onTimeoutWarning.subscribe(
+			(countdown: number) => {
+				if (countdown === this.idle.getTimeout()) {
+					this.countdownStarted = true;
+					// dispatch action so an effect can run any logic if needed (i.e. displaying a timeout countdown dialog)
+					this.store.dispatch(new SessionTimeoutCountdownStart(countdown));
+				}
+			}
+		);
+	}
+
+	public configureKeepaliveService(): void {
+		if (!this.keepalive) {
+			return;
+		}
+
+		this.keepalive.interval(this.appConfig.keepAliveInterval); // ping interval in seconds
+
+		let pingRequestHeaders: HttpHeaders = new HttpHeaders();
+		pingRequestHeaders = pingRequestHeaders.set(StarkHttpHeaders.NBB_CORRELATION_ID, this.logger.correlationId);
+
+		// FIXME Where does DEVELOPMENT Variable come from ???
+		// if (DEVELOPMENT) {
+			this.fakePreAuthenticationHeaders.forEach((value: string, key: string) => {
+				pingRequestHeaders = pingRequestHeaders.set(key, value);
+			});
+		// }
+		
+		if (typeof this.appConfig.keepAliveUrl !== "string") {
```

Comment:
This if block is not necessary for 2 reasons:

1.- the keepAliveUrl has a class validation (IsURL) so this will never be diferent than a string
2.- in the constructor of this service we validate the config so an undefined keepAliveUrl will be caught there meaning that the application will throw an error before reaching this code